test/unit/queue/util/createProducer-test.js

```diff
@@ -109,3 +112,30 @@ test('should throw an error if a producer cannot start', async (t) => {
 
   t.pass()
 })
+
+test('should emit an error if connection emits an error', async (t) => {
+  const { connection } = t.context
+
+  const producer = await createProducer({
+    logger: console,
+    amqUrl: TEST_AMQ_URL,
+    producerOptions: {
+      queueName: TEST_QUEUE_NAME,
+      connection
+    }
+  })
+
+  const errorEventsPromise = Promise.all([
+    waitForEvent(producer, 'error', (err) => {
+      t.is(err.message, 'test error')
+      return true
+    }),
+    waitForEvent(connection, 'error', (err) => {
+      t.is(err.message, 'test error')
+      return true
+    })
```

Comment:
Alternatively you could just return `err.message === 'test error'` for both filter functions